server/pulp/server/async/app.py

```diff
@@ -72,7 +166,9 @@ def initialize_worker(sender, instance, **kwargs):
     initialization.initialize()
 
     # Delete any potential old state
-    tasks._delete_worker(sender, normal_shutdown=True)
+    existing_worker = Worker.objects(name=sender).first()
+    if existing_worker:
```

Comment:
This is susceptible to a race condition. Rather than checking and doing something conditional based on that, try to have the all of the work occur in the database. If you want a more specific idea let me know.